plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -127,7 +127,20 @@ func (g *genericScheduler) Schedule(pod *v1.Pod, nodeLister algorithm.NodeLister
 	}
 
 	trace.Step("Selecting host")
-	return g.selectHost(priorityList)
+	host, err := g.selectHost(priorityList)
+	if err != nil {
+		return "", err
+	}
+
+	if len(g.extenders) != 0 {
+		for _, extender := range g.extenders {
+			err := extender.Bind(pod, host)
```

Comment:
I'd suggest to implement `Binder` interface (let extender bind to apiserver) and replace current default binder (bind to apiserver) according to configuration. There's a case that: extender.Bind succeed but bind to apiserver failed. The extender did not how to recover it. If extender sync up periodically, there'll be a race condition with scheduler's assumed timeout.